packages/web-components/src/avatar/avatar.ts

```diff
@@ -0,0 +1,175 @@
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr
+  public name: string | undefined;
+  protected nameChanged(previous: string, next: string): void {
+    if (next !== undefined && !this.initials) {
+      this.setInitials();
+    }
+  }
+
+  /**
+   * Provide custom initials rather than one generated via the name
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr
+  public initials: string | undefined;
+  protected initialsChanged(previous: string | undefined, next: string | undefined): void {
+    if (next !== undefined) {
+      this.setInitials();
+    }
+  }
+
+  /**
+   * Size of the avatar in pixels.
+   *
+   * Size is restricted to a limited set of supported values recommended for most uses (see `AvatarSizeValue`) and
+   * based on design guidelines for the Avatar control.
+   *
+   * If a non-supported size is neeeded, set `size` to the next-smaller supported size, and set `width` and `height`
+   * to override the rendered size.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   *
+   */
+  @attr({ converter: nullableNumberConverter })
+  public size: AvatarSize | undefined;
+
+  /**
+   * The avatar can have a circular or square shape.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: shape
+   */
+  @attr
+  public shape: AvatarShape | undefined;
+
+  /**
+   * Optional activity indicator
+   * * active: the avatar will be decorated according to activeAppearance
+   * * inactive: the avatar will be reduced in size and partially transparent
+   * * undefined: normal display
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: active
+   */
+  @attr
+  public active?: AvatarActive | undefined;
+
+  /**
+   * The appearance when `active="active"`
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: appearance
+   */
+  @attr
+  public appearance?: AvatarAppearance | undefined;
+
+  /**
+   * The color when displaying either an icon or initials.
+   * * neutral (default): gray
+   * * brand: color from the brand palette
+   * * colorful: picks a color from a set of pre-defined colors, based on a hash of the name (or colorId if provided)
+   * * [AvatarNamedColor]: a specific color from the theme
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: color
+   */
+  @attr
+  public color: AvatarColor = 'neutral';
+  protected colorChanged(previous: string, next: string): void {
+    this.setColor();
+  }
+
+  /**
+   * Specify a string to be used instead of the name, to determine which color to use when color="colorful".
+   * Use this when a name is not available, but there is another unique identifier that can be used instead.
+   */
+  @attr({ attribute: 'color-id' })
+  public colorId: AvatarNamedColor | undefined;
+  public colorIdChanged(): void {
+    if (!this.colorId) {
+      return;
+    }
+
+    this.setColor();
+  }
+
+  /**
+   * Sets the data-color attribute used for the visual presentation
+   * @internal
+   */
+  public setColor(): AvatarColor | void {
```

Comment:
@miroslavstastny happy to update to a better semantic, what would you suggest for clarity? FWIW, `setColor` is necessary and we're using the data attribute because colorful is "side-effect-ish"...so, the attribute for color isn't reliable for styles...perhaps that'll help pull out intent and semantics. Would `generate` be more appropriate?

Reply:
I went ahead and changed these to `generate____()` because in the case of initials we're creating them based on an input of some sort and in the case of color we are generating the value for the data attribute based on the color selection (given colorful means we generate a given color). It seems more semantic to me, let me know if you have an alternative preference though.